she_codes_news/news/views.py

```diff
@@ -1,16 +1,59 @@
 from django.views import generic
+from django.urls import reverse_lazy
+from django.shortcuts import get_object_or_404
 from .models import NewsStory
+from .forms import StoryForm, CommentForm
 
 
 class IndexView(generic.ListView):
     template_name = 'news/index.html'
+    context_object_name = 'all_stories'
 
     def get_queryset(self):
         '''Return all news stories.'''
-        return NewsStory.objects.all()
+        qs = NewsStory.objects.all()
+        sort = self.request.GET.get("sort")
+        print (f"{sort=}")
+        if sort == "OldestFirst":
+            qs = qs.order_by('pub_date')
+        else:
+            qs = qs.order_by('-pub_date')
+        return qs
+        
+        
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['latest_stories'] = NewsStory.objects.all()[:4]
-        context['all_stories'] = NewsStory.objects.all()
         return context
+
+class StoryView(generic.DetailView):
+    model = NewsStory
+    template_name = 'news/story.html'
+    context_object_name = 'story'
+ 
+
+class AddStoryView(generic.CreateView):
+    form_class = StoryForm
+    context_object_name = 'storyForm'
+    template_name = 'news/createStory.html'
+    success_url = reverse_lazy('news:index')
+
+    def form_valid(self, form):
+        form.instance.author = self.request.user
+        return super().form_valid(form)
+
+class AddCommentView(generic.CreateView):
+    form_class = CommentForm
```

Comment:
You could possibly have skipped creating a ModelForm for this, and just used the `fields` option for create view, which would have automagically created a form for you!